### DISCUSSION TITLE
Grid columns go on top of each other

### DISCUSSION THREAD

**Discussion by stackerito** [tailwindlabs/tailwindcss#5118] [Help] [answered]
I have the following code:
```
<div class="h-screen w-64 bg-gray-300">
        <aside class="grid grid-cols-2 bg-yellow-400 h-full">

            <div class="bg-red-300 w-64 opacity-50">
                One
            </div>
            <div class="bg-green-300 w-64">
                Two
            </div>
        </aside>
    </div>
```
Demo; https://codepen.io/Stackerito/pen/abWGbXm

The 2 inner divs are the same width as the parent div on purpose - I am planning to use `overflow-x-hidden` to show only one div at a time, while pushing the other div away and hidden. But they just go on top of each other instead of being side by side.

How can I fix that and make it work with `overflow-x-hidden`? Because when I tried to set width to the `aside`, then the `overflow-hidden-x` didn't work

**Comment by simonswiss** [comment, marked as answer]
Hey!

I don't think CSS grid is the right tool for the job here. When you define `grid-cols-2` on the aside, the resulting CSS is this:

```grid-template-columns: repeat(2, minmax(0, 1fr));```

That creates `2` gird columns, which will each take up an amount of space that ranges from `0` (min) to `1fr`, or one _fraction_ of the space available (max).

In this case, both columns take up `1fr`, and since there are two columns, each `fr` is half the space of the container. The width applied with `w-64` is applied, but not respected since the grid container gets to decide here.

You're better off using another layout approach here – flex, inline-block..

Here's an example of your Play using flex instead:

https://play.tailwindcss.com/iIitLaoGS0

https://user-images.githubusercontent.com/485747/127591189-41b8ae83-a025-48a8-96a1-a6a1e67bd679.mov

Instead of `grid grid-cols-2`, I use `flex` on the `<aside>` element.

That alone would have the same result, splitting the space in two, but I add a `flex-shrink-0` to each child, so they are "not allowed" to shrink, and will respect the `w-64` width.

As a result, they will take 2x the space of the container, but adding `overflow-x-auto` on the container will make the container scrollable horizontally, which I _think_ is what you want.

Hope it helps!

**Comment by stackerito** [comment]
Yes, this is the end result I'm looking for (of course I'll need to use JS to scroll between the two).
But, as you said since grid uses `1fr` I decided to add a fixed width to the `<aside>` and then add `overflow-x-hidden` to the parent wrapping div. This way, when using grid, I can add as much columns as I want, in the following example I made 3 columns instead of two. Can you please take a look and say if there's something wrong with it (That might cause problems in the future):
https://play.tailwindcss.com/npWfeoJC46

**Reply by simonswiss** [reply]
I sitll don't think `grid` is the best tool here.

I mean, you'd have to dynamically update the grid parent's width based on the number of children? `w-32` for 1 child, `w-64` for 2 children etc.

Using flex + `flex-shrink-0` on each child, no matter how many "columns" or elements you have, you won't have to set the parent width to the total width of the cumulated children:

https://play.tailwindcss.com/UMYEZYFgy3

Seems like a good enough reason to go with flex over grid in this particular situation, in my opinion.

There is nothing _wrong_ with using grid by the way, just sounds like you're doing extra work to make sure the parent determined what "hidden" width it should have so the children look the correct width.
